### PR TITLE
Drupal to drupal replication

### PR DESCRIPTION
**Pull request by jeqq**
I will rebase this branch and squash a part of commits after PR #22 will be merged.

### CODE REVIEW

**Comment by dickolsson** [comment]
@jeqq Nice! Awesome that we can use the CouchDB replicator to replicate between two Drupal sites! I will review this now...

**Comment by dickolsson** [comment]
Why do we need two vhost files? Can we not use one template and just run `cp` and `sed` for both `drupal1` and `drupal2`?

Also, instead of doing `sed /DocumentRoot/i\ServerName` I think we should add a new line in the vhost template that looks like this:

`ServerName %TRAVIS_BUILD_SERVERNAME%`

Then we can just do `sudo sed -e "s?%TRAVIS_BUILD_SERVERNAME%?drupal2?g" --in-place [vhost-file]`. I think that's cleaner.

**Comment by jeqq** [comment]
I used two files because this line `FastCgiExternalServer /usr/lib/cgi-bin/php5-fcgi -host 127.0.0.1:9000 -pass-header Authorization` should be used just for default vhost. I will remove one file an will remove this line with `sed`.

**Comment by dickolsson** [comment]
@jeqq Ok, sounds good for now.

File: src/Tests/Encoder/StreamEncoderTest.php [hunk 1]
```diff
@@ -12,6 +12,8 @@
  */
 class StreamEncoderTest extends KernelTestBase {
 
+  protected $strictConfigSchema = FALSE;
```
Comment (dickolsson) [inline src/Tests/Encoder/StreamEncoderTest.php:15]: What's the reason why we need to use this in some tests?

Comment (jeqq) [inline src/Tests/Encoder/StreamEncoderTest.php:15]: We have a schema error for PUT and HEAD methods when running tests - methods that are not defined by Rest module. Setting this to TRUE just avoid the check in `Drupal\Core\Config\Testing\ConfigSchemaChecker::onConfigSave()`.

**Comment by dickolsson** [comment]
@jeqq I have now merged #22, please rebase this branch as you mentioned.

**Comment by dickolsson** [comment]
Awesome work @jeqq!

**Pull request merged**